### PR TITLE
feat(TDI-45155) : mscrm dynamics expand

### PR DESCRIPTION
**Pull request by ypiel-talend**
**What is the current behavior?** (You can also link to an open issue here)
No support of odata expandable attributes in ms crm.

**What is the new behavior?**
Support of expandable attribute in ms crm.
https://jira.talendforge.org/browse/TDI-45155

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.localprovider/components/tMicrosoftCrmInput/tMicrosoftCrmInput_begin_odata.javajet [hunk 1]
```diff
@@ -200,11 +229,45 @@ if ((metadatas != null) && (metadatas.size() > 0)) {
...
+	                      row1.<%=expandRow.get("OUTPUT_COLUMN")%> = new java.io.BufferedReader(new java.io.InputStreamReader(exp_entity_<%=in_col%>_inputStream_<%=cid%>)).lines().collect(java.util.stream.Collectors.joining("\n"));
+	                  }
+	                  {
+	                      final org.apache.olingo.client.api.domain.ClientInlineEntitySet clientInlineEntitySet_<%=in_col%>_<%=cid%> = expandLinks_tMicrosoftCrmInput_1.get(sanitizedName_<%=in_col%>_<%=cid%>).asInlineEntitySet();
+	                      if(clientInlineEntitySet_<%=in_col%>_<%=cid%> != null){
+	                          final java.io.InputStream exp_entity_<%=in_col%>_inputStream_<%=cid%> = client_tMicrosoftCrmInput_1.getClient().getWriter().writeEntities(clientInlineEntitySet_<%=in_col%>_<%=cid%>.getEntitySet().getEntities(), org.apache.olingo.commons.api.format.ContentType.APPLICATION_JSON);
+	                          row1.<%=expandRow.get("OUTPUT_COLUMN")%> = new java.io.BufferedReader(new java.io.InputStreamReader(exp_entity_<%=in_col%>_inputStream_<%=cid%>)).lines().collect(java.util.stream.Collectors.joining("\n"));
```
Comment (olhava) [inline main/plugins/org.talend.designer.components.localprovider/components/tMicrosoftCrmInput/tMicrosoftCrmInput_begin_odata.javajet:252]: Is it possible to use try with resources here?

File: main/plugins/org.talend.designer.components.localprovider/components/tMicrosoftCrmInput/tMicrosoftCrmInput_java.xml [hunk 2]
```diff
@@ -27544,7 +27551,7 @@
                 <IMPORT NAME="jcifs" MODULE="jcifs-1.3.0.jar" MVN="mvn:org.talend.libraries/jcifs-1.3.0/6.0.0"  REQUIRED_IF="((AUTH_TYPE == 'ON_PREMISE') AND (MS_CRM_VERSION == 'CRM_2011')) OR (API_VERSION=='API_2007')" />
                 <!-- 2011 -->
                 <!-- crm client -->
-                <IMPORT NAME="talend-mscrm" MODULE="talend-mscrm-3.4-20210428.jar" MVN="mvn:org.talend.components/talend-mscrm/3.4-20210428" REQUIRED_IF="(((AUTH_TYPE=='ONLINE') AND (API_VERSION=='API_2011' OR API_VERSION =='API_2016_ODATA' OR API_VERSION =='API_2018_ODATA')) OR ((AUTH_TYPE=='ON_PREMISE') AND (MS_CRM_VERSION == 'CRM_2016' OR MS_CRM_VERSION == 'CRM_2018')))" />
+                <IMPORT NAME="talend-mscrm" MODULE="talend-mscrm-3.5-20210518.jar" MVN="mvn:org.talend.components/talend-mscrm/3.5-20210518"  UrlPath="platform:/plugin/org.talend.libraries.crm/lib/talend-mscrm-3.4-20210518.jar" REQUIRED_IF="(((AUTH_TYPE=='ONLINE') AND (API_VERSION=='API_2011' OR API_VERSION =='API_2016_ODATA' OR API_VERSION =='API_2018_ODATA')) OR ((AUTH_TYPE=='ON_PREMISE') AND (MS_CRM_VERSION == 'CRM_2016' OR MS_CRM_VERSION == 'CRM_2018')))" />
```
Comment (olhava) [inline main/plugins/org.talend.designer.components.localprovider/components/tMicrosoftCrmInput/tMicrosoftCrmInput_java.xml:27554]: Version should be bumped also for tMicrosoftCrmOutput_java?

File: main/plugins/org.talend.designer.components.localprovider/components/tMicrosoftCrmInput/tMicrosoftCrmInput_begin_odata.javajet [hunk 3]
```diff
@@ -150,6 +151,31 @@ if ((metadatas != null) && (metadatas.size() > 0)) {
...
+                                        expand_incol_<%=cid%>.add("<%=expandRow.get("INPUT_COLUMN")%>");
+                                        expand_params_<%=cid%>.add("<%=expandRow.get("EXPAND_PARAMS")%>");
+                                    <%
+                                }
+                                %>
+                                        expands_<%=cid%> = org.talend.ms.crm.odata.QueryExpandUtil.getExpandConfig(expand_incol_<%=cid%>, expand_params_<%=cid%>);
+                                <%
+                        }
+			java.util.List<String> expandOutColNames = new java.util.ArrayList();
+			for(java.util.Map<String, String> expand_row : expandArray){
```
Comment (olhava) [inline main/plugins/org.talend.designer.components.localprovider/components/tMicrosoftCrmInput/tMicrosoftCrmInput_begin_odata.javajet:174]: Need to fix expand_row name according to the conventions

**Review by ypiel-talend — COMMENTED** [review]
Thanks for good remarks :+1: 
- I use try-with-resource
- I bump lib version in tMSCrmOuput
- I update variable name

**Review by olhava — APPROVED** [review]

**Review by olhava — APPROVED** [review]

**Review by olhava — DISMISSED** [review]

**Review by olhava — APPROVED** [review]

**Review by olhava — APPROVED** [review]

**Pull request merged**